viewing_party/party.py

```diff
@@ -1,23 +1,270 @@
+    and returns a string that represents that genre
+    '''
+    watched_genre_count = {}
+    most_watched_genre_total = 0
+    most_watched_genre = ""
+
+    user_data_watched = user_data["watched"]
+
+    if user_data_watched == []:
+        return None
+
+    for movie in user_data_watched:
+        if movie["genre"] in watched_genre_count:
+            watched_genre_count[movie["genre"]] += 1
+        else:
+            watched_genre_count[movie["genre"]] = 1
+
+    for genre in watched_genre_count:
+        if watched_genre_count[genre] > most_watched_genre_total:
+            most_watched_genre_total = watched_genre_count[genre]
+            most_watched_genre = genre
+
+    return most_watched_genre
+
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    '''
+    collects uniqe movies watched by user compared to friends and
+    stores it in a list, then returns that list
+    '''
+    unique_watched_list = []
+
+    friend_watched_set = set(get_friends_watched(user_data))
+    user_watched_set = set(get_users_watched(user_data))
+
+    unique_set = user_watched_set - friend_watched_set
+
+    for i in range(len(user_data["watched"])):
+        if user_data["watched"][i]['title'] in unique_set:
+            unique_watched_list.append(user_data["watched"][i])   
+
+    return unique_watched_list
+    
+    
+    
+def get_friends_watched(user_data):
+    '''
+    Helper Function - Stores the titles of the friends 'watched' list from 
+    user_data in a list, and returns that list
+    '''   
+    friends_list_of_dict = user_data["friends"]
+    friends_watched_list = []
+
+    for i in range(len(friends_list_of_dict)):
+        for watched in friends_list_of_dict[i]:
+            for movie in friends_list_of_dict[i][watched]:
+                friends_watched_list.append(movie['title'])
+
+    return friends_watched_list
+
+def get_users_watched(user_data):
+    '''
+    Helper Function - Stores the titles of the users 'watched' list from 
+    user_data in a list, and returns that list
+    '''      
+    
+    user_watched_list = user_data["watched"]
+    users_watched_movies = []
+
+    for movie in user_watched_list:
+        users_watched_movies.append(movie['title'])
+
+    return users_watched_movies
+
+
+def get_friends_unique_watched(user_data):
+    '''
+    collects unique watched movies from friends, stores them in a list,
+    and returns the list
+    '''
+    unique_watched_list = []
+
+    # User Data Watched Sets
+    friend_watched_set = set(get_friends_watched(user_data))
+    user_watched_set = set(get_users_watched(user_data))
+
+    unique_set = friend_watched_set - user_watched_set
+
+    # User Data Variables
+    friends_list_dict = user_data["friends"]
+
+
+    for i in range(len(friends_list_dict)):
+        for friend_movie in friends_list_dict[i]["watched"]:

```

Comment:
This style of looping works, but a more "pythonic" style of loop here would be to use the `for item in list` pattern:
```suggestion
    for friend in friends_list_dict:
        for friend_movie in friend["watched"]:

```